Userland/shutdown.cpp

```diff
@@ -8,8 +9,10 @@ int main(int argc, char** argv)
     CArgsParserResult args = args_parser.parse(argc, argv);
 
     if (args.is_present("n")) {
-        syscall(SC_halt);
-        return 0;
+        if (halt() < 0) {
```

Comment:
Also cleaner, thankyou for making my mess better :)

Reply:
Hahaha nw. Usually I'm the one writing horribly messy code (as is evidenced by my PRs :P)